### PR TITLE
Add project filter to fuzzer stats.

### PR DESCRIPTION
**Pull request by inferno-chromium**
Also, remove route hacks and instead use query params.

### CODE REVIEW

File: src/appengine/handlers/fuzzer_stats.py [hunk 1]
```diff
@@ -387,43 +387,68 @@ def _get_date(date_value, days_ago):
...
-    user_email = helpers.get_user_email()
+    if not access.has_access():
+      # User is an external user of ClusterFuzz (eg: non-Chrome dev who
+      # submitted a fuzzer or someone with a project in OSS-Fuzz).
+      user_email = helpers.get_user_email()
+      fuzzers_list = external_users.allowed_fuzzers_for_user(
+          user_email, include_from_jobs=True, include_parents=True)
+      if not fuzzers_list:
+        # User doesn't actually have access to any fuzzers.
+        raise helpers.AccessDeniedException()
```
Comment (oliverchang) [inline src/appengine/handlers/fuzzer_stats.py]: nit: add a message?

File: src/appengine/handlers/fuzzer_stats.py [hunk 2]
```diff
@@ -387,43 +387,68 @@ def _get_date(date_value, days_ago):
...
+      fuzzers_list = sorted(
+          external_users.allowed_fuzzers_for_user(
+              user_email, include_from_jobs=True, include_parents=True))
       if not fuzzers_list:
         # User doesn't actually have access to any fuzzers.
         raise helpers.AccessDeniedException()
 
-      jobs_list = external_users.allowed_jobs_for_user(user_email)
+      jobs_list = sorted(external_users.allowed_jobs_for_user(user_email))
+      projects_list = sorted(
```
Comment (oliverchang) [inline src/appengine/handlers/fuzzer_stats.py]: Do we need to filter out duplicates?

File: src/appengine/handlers/fuzzer_stats.py [hunk 3]
```diff
@@ -387,43 +387,68 @@ def _get_date(date_value, days_ago):
...
+              include_parents=True, project=project))
+      jobs_list = data_handler.get_all_job_type_names(project=project)
     else:
       # User is an external user of ClusterFuzz (eg: non-Chrome dev who
       # submitted a fuzzer or someone with a project in OSS-Fuzz).
-      fuzzers_list = external_users.allowed_fuzzers_for_user(
-          user_email, include_from_jobs=True, include_parents=True)
+      user_email = helpers.get_user_email()
 
+      fuzzers_list = sorted(
```
Comment (oliverchang) [inline src/appengine/handlers/fuzzer_stats.py]: Is it intentional that project filters aren't supported here? (e.g. if someone external has access to more than 1).

File: src/appengine/handlers/fuzzer_stats.py (hunk 2 above)
Comment (inferno-chromium) [reply src/appengine/handlers/fuzzer_stats.py]: yes right, forgot about it.

File: src/appengine/handlers/fuzzer_stats.py (hunk 1 above)
Comment (inferno-chromium) [reply src/appengine/handlers/fuzzer_stats.py]: Done!

File: src/appengine/handlers/fuzzer_stats.py (hunk 3 above)
Comment (inferno-chromium) [reply src/appengine/handlers/fuzzer_stats.py]: yes added a TODO. I couldn't figure out how to support this properly, i think a fix has to come in external_users functions, wdyt ?

**Pull request merged**